pimd/pim_upstream.c

```diff
@@ -619,6 +621,8 @@ static struct pim_upstream *pim_upstream_new(struct pim_instance *pim,
 
 		hash_release(pim->upstream_hash, up);
 		XFREE(MTYPE_PIM_UPSTREAM, up);
+		if (ch)
```

Comment:
so this is a good catch, in not resetting the ch->upstream, but we just accepted an patch that allows upstream creation when the RP is not there.  Do those patches solve this problem a bit differently?